core/silkworm/execution/evm.cpp

```diff
@@ -214,8 +214,23 @@ evmc::result EVM::call(const evmc_message& message) noexcept {
 evmc::result EVM::execute(const evmc_message& msg, ByteView code, std::optional<evmc::bytes32> code_hash) noexcept {
     address_stack_.push(msg.destination);
 
-    EvmHost host{*this};
-    evmc_revision rev{revision()};
+    evmc_result res;
+    if (exo_evm) {
+        EvmHost host{*this};
+        res = exo_evm->execute(exo_evm, &host.get_interface(), host.to_context(), revision(), &msg, code.data(),
+                               code.size());
+    } else {
+        res = execute_endogenously(msg, code, code_hash);
+    }
+
+    address_stack_.pop();
+
+    return evmc::result{res};
+}
+
+evmc_result EVM::execute_endogenously(const evmc_message& msg, ByteView code,
+                                      std::optional<evmc::bytes32> code_hash) noexcept {
+    const evmc_revision rev{revision()};
```

Comment:
Why not pass `rev` as parameter so that it is clear in `EVM::execute` that both use the same revision?

Reply:
OK, done